### PR TITLE
Freeipa create users

### PR DESCRIPTION
**Pull request by DavidFair**
Adds a workflow and supporting action(s) to create users on a given FreeIPA server.

### CODE REVIEW

File: actions/workflow.freeipa.user.create.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+description: Creates a given list of users
+enabled: true
+entry_point: workflows/freeipa.user.create.yaml
+runner_type: orquesta
+
+parameters:
+  submodule:
+    default: user_delete
+    immutable: true
+    type: string
```
Comment (joelvdavies) [inline actions/workflow.freeipa.user.create.yaml:11]: This should not be here - its not taken in by the workflow.

File: actions/workflow.freeipa.user.create.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  server_env:
+    description: The FreeIPA environment to create users on
+    required: true
+    type: string
+    enum:
+      - prod
+      - dev
+  user_base_name:
+    description: The users to remove, or base name of the users to remove without the hyphen (e.g. "jupyter")
```
Comment (joelvdavies) [inline actions/workflow.freeipa.user.create.yaml:20]: This description isn't for creating users.

File: actions/workflow.freeipa.user.create.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    required: true
+    type: string
+  first_index:
+    description: The index of the first user to remove
```
Comment (joelvdavies) [inline actions/workflow.freeipa.user.create.yaml:24]: ^

File: actions/workflow.freeipa.user.create.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    required: false
+    type: integer
+  last_index:
+    description: The index of the last user to remove (inclusive).
```
Comment (joelvdavies) [inline actions/workflow.freeipa.user.create.yaml:28]: ^

**Review by joelvdavies — CHANGES_REQUESTED** [review]
A few things that weren't changed after what I assume was a copy & paste.

File: actions/workflow.freeipa.user.create.yaml (hunk 1 above)
Comment (ElliottKasoar) [reply actions/workflow.freeipa.user.create.yaml:11]: Should also be `name: workflow.freeipa.user.create` to be consistent with other `workflow.[x]` actions?

Comment (joelvdavies) [reply actions/workflow.freeipa.user.create.yaml:11]: @ElliottKasoar Yes I was in two minds about it, on the one hand they are all actions that happen to run workflows and freeipa.user.create would probably be the expected name for this but it is inconsistent, so I think your right it should be workflow.* for now at least.

**Review by joelvdavies — APPROVED** [review]
LGTM

**Pull request merged**